packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -46,10 +47,10 @@ class BuildInfo {
   final bool trackWidgetCreation;
 
   /// Extra command-line options for front-end.
-  final String extraFrontEndOptions;
+  final List<String> extraFrontEndOptions;
```

Comment:
These don't seem to be used anywhere, but fixing them is out of scope for this change. Using `getBuildInfo` in more commands revealed that the types of the argument + these values did not always match up. Some commands had specified this as a multi-option, but that returns a List<String> so I tweaked the build info to keep it in that format

Reply:
Are the Dart defines a special case of front end options? Or do they go different places?

Reply:
Kind of, we currently need to handle them different for dart2js builds so sticking them in the same list is not really an advantage. Furthermore, the extraFrontEndOptions and extraGenSnapshotOptions are actually broken and don't go anywhere for now.